src/plugins/intel_npu/src/plugin/npuw/serialization.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2024-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "serialization.hpp"
+
+#include "intel_npu/config/config.hpp"
+#include "logging.hpp"
+#include "openvino/op/constant.hpp"
+#include "spatial.hpp"
+
+void ov::npuw::s11n::write(std::ostream& stream, const std::streampos& var) {
+    stream.write(reinterpret_cast<const char*>(&var), sizeof var);
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const std::string& var) {
+    auto var_size = var.size();
+    stream.write(reinterpret_cast<const char*>(&var_size), sizeof var_size);
+    stream.write(&var[0], var.size());
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const bool& var) {
+    stream.write(reinterpret_cast<const char*>(&var), sizeof var);
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const ov::npuw::compiled::Spatial& var) {
+    using ov::npuw::s11n::write;
+
+    write(stream, var.params.size());
+    for (const auto& p : var.params) {
+        write(stream, p.idx);
+        write(stream, p.dim);
+    }
+    write(stream, var.range);
+    write(stream, var.nway);
+    write(stream, var.out_dim);
+    write(stream, var.nway_iters);
+    write(stream, var.tail_size);
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const ov::Tensor& var) {
+    using ov::npuw::s11n::write;
+
+    NPUW_ASSERT(var.is_continuous());
+    auto type_str = var.get_element_type().to_string();
+    write(stream, type_str);
+    write(stream, var.get_shape());
+    // FIXME: Should strides be serialized as well?
+    write(stream, var.get_byte_size());
+    stream.write(reinterpret_cast<const char*>(var.data()), var.get_byte_size());
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const ::intel_npu::Config& var) {
+    write(stream, var.toString());
+}
+
+void ov::npuw::s11n::write(std::ostream& stream, const ov::Output<const ov::Node>& var) {
+    write(stream, var.get_element_type().to_string());
+    write(stream, var.get_partial_shape().to_string());
+    write(stream, var.get_names());
+}
```

Comment:
uhm, do we really serialize something like this? it may be quite misleading in the code.

Reply:
We do to create a dummy ov::Model object for `CompiledModel` constructor

Reply:
That's right. Do I need to do anything about this comment?

Reply:
I think we need to rethink our serialization overall, the code can be reduced by 2X - put let's keep it for future